Sns launch fixes (#1611)
dfinity/portal: docs/developer-docs/integrations/sns/launching/launch-steps.md

# SNS launch steps

## Overview
An SNS can be launched in production by following the steps explained on a 
high level [here](../launching/launch-summary.md).

Technically, these are the same steps that the
[SNS local testing repository](../testing/local-testing.md) guides you through,
with the difference that the commands target the canisters on the mainnet.

To make the most important commands and what they need to look like for 
mainnet more accessible, they are listed below.

## Requirememts

1. IC SDK

2. `ic-admin`

3. `sns` CLI

:::note
The version of the sns CLI that is bundled with your dfx version may not have the latest commands described in the Usage section. If needed, it is recommended to build and use the sns CLI tool yourself.
:::

```bash
git clone git@github.com:dfinity/ic.git
cd ic
bazel build //rs/sns/cli:sns
ls bazel-bin/rs/sns/cli/sns 
```

## Steps

- #### Step 1: Dapp developers choose the initial parameters of the SNS for a dapp

> - #### Step 2: Dapp developers submit NNS proposal so they can deploy to the SNS subnet

Reviewer: Note that I think in the past we deliberately did not specify who needs to do this as this is really a NNS DAO decision, so everyone who is eligible on the NNS can send this proposal.
I don't have a strong preference here but wonder if it would be good to make sure this is aligned with language from Pete + co.

> Anyone who owns and NNS neuron with enough stake can submit a proposal

Reviewer: ```suggestion
Anyone who owns an NNS neuron with enough stake can submit a proposal
```

that lists a principal wallet in [SNS-W](../introduction/sns-architecture.md#SNS-W) who can then deploy the SNS canisters.

To create such a proposal, a common path is to [download the `ic-admin` tool](../../../setup/ic-admin.md) and run the following:

```bash 
ic-admin  \
   --nns-url "${NETWORK_URL}" propose-to-update-sns-deploy-whitelist  \
   --added-principals "${WALLET}"  \
   --proposal-title "Let me SNS!"  \
   --summary "This proposal whitelists developer's principal to deploy SNS"
``` 

* One can substitute `WALLET` with the principal in question 
* One can substitute `NETWORK_URL` with `https://nns.ic0.app`

- #### Step 3: NNS DAO accepts or rejects the NNS proposal

>   This is the **first of three** proposals that need to successfully pass.

Reviewer: ```suggestion
  This is the **first of two NNS** proposals that need to successfully pass.
```
Maybe like this? I think the SNS proposals are in another category.

  
  :::info

>   If this NNS proposal passes and the developer's principal is added the list of principals that can deploy to the SNS subnet, it does **not** guarantee the rest of the steps will complete.

Reviewer: ```suggestion
  If this NNS proposal passes and the developer's principal is added to the list of principals that can deploy to the SNS subnet, it does **not** guarantee the rest of the steps will complete.
```

  :::

- #### Step 4: Dapp developers trigger the SNS canisters to be created on SNS subnet

After the wallet canister is listed in SNS-W, 
the [SNS canisters are created triggered by a manual call to SNS-W](../launching/launch-steps.md/#SNS-launch-step-deployment).

> The command tp trigger the creation of SNS canisters on the SNS subnet: 

Reviewer: ```suggestion
The command to trigger the creation of SNS canisters on the SNS subnet: 
```
